src/utils/ctl/cart_ctl.c

```diff
@@ -734,7 +734,8 @@ main(int argc, char **argv)
 {
 	int		rc = 0;
 
-	d_log_init();
+	rc = d_log_init();
+	D_ASSERTF(rc == 0, "d_log_init failed, rc=%d\n", rc);
```

Comment:
Asserts don't seem like a particuarly friendly way to abort cart_ctl?

Reply:
It is used both ways in the code. Since this is a utility I used the method used in the crt_utils however I can make it a return instead. However we also assert if parse_args fails as well. So we already seem to use an assert in that code if the startup code fails.

Reply:
With respect to the assert in server_iv.c that was the suggested fix given by Di. I can go back to do an if statement up top with an error code but he suggested to use the asserts instead since we should never have those op codes with a null sync.

Reply:
Yeah, it wasn't really an indictment of this patch, just an observation. I think an assert is fine, just making a comment that someone might want to clean up the error handling for the overall utility.  If it's just an internal utility, it's fine but if any users use it, an assert isn't the best way to notify them that they typed in a bad parameter.